src/main/java/ch/njol/skript/sections/recipe/SecCookingRecipe.java

```
@@ -0,0 +1,192 @@
+		return "create crafting recipe for " + result.toString(event, debug) + " with key " + key.toString(event, debug);
+	}
+
+	@Override
+	protected TriggerItem walk(Event event) {
+		execute(event);
+		return walk(event, false);
+	}
+
+	private void execute(Event event) {
+		String key = this.key.getSingle(event);
+		if (key == null)
+			return;
+		ItemType result = this.result.getSingle(event);
+		if (result == null)
+			return;
+		ItemType ingredient = this.ingredient.getSingle(event);
+		if (ingredient == null)
+			return;
+		Number xp = this.xp.getSingle(event);
+		if (xp == null)
+			return;
+		Timespan cookTime = this.cookTime.getSingle(event);
+		if (cookTime == null)
+			return;
+
+		Set<Material> ingredientMaterials = new HashSet<>();
+		for (ItemStack itemStack : ingredient.getAll()) {
+			ingredientMaterials.add(itemStack.getType());
+		}
+
+		// the recipe APIs require an int :(
+		int cookTimeTicks = (int) cookTime.getTicks_i();
+		NamespacedKey namespacedKey = Utils.getNamespacedKey(key);
+		RecipeChoice choice = new RecipeChoice.MaterialChoice(ingredientMaterials.toArray(new Material[0]));
+		CookingRecipe recipe;
+		switch (type) {
+			case SMOKER:
+				recipe = new SmokingRecipe(namespacedKey, result.getRandom(), choice, xp.floatValue(), cookTimeTicks);
+				break;
+			case FURNACE:
+				recipe = new FurnaceRecipe(namespacedKey, result.getRandom(), choice, xp.floatValue(), cookTimeTicks);
+				break;
+			case CAMPFIRE:
+				recipe = new CampfireRecipe(namespacedKey, result.getRandom(), choice, xp.floatValue(), cookTimeTicks);
+				break;
+			case BLAST_FURNACE:
+				recipe = new BlastingRecipe(namespacedKey, result.getRandom(), choice, xp.floatValue(), cookTimeTicks);
+				break;
+			default:
+				throw new IllegalStateException();
+		}
+
+		try {
+			Bukkit.getServer().addRecipe(recipe);
+		} catch (IllegalStateException ignored) {
+			// Bukkit throws a IllegalStateException if a duplicate recipe is registered
+		}
```

Could we not just remove the duplicated recipe? I find that better than ignoring the error together

---

Configuration option to override is a possibility.

---

> Could we not just remove the duplicated recipe? I find that better than ignoring the error together

I'd have to write something that manually compares recipes to find duplicates. I would rather just rely on Bukkit's logic for detecting dupes so that bukkit is "in charge" of determining what is and isn't a duplicate recipe.